### ISSUE TITLE
disabling docker pull/push to docker hub

### ISSUE THREAD

**Issue by CharanBR** [open-policy-agent/opa-docker-authz#23]
@tsandall,

Is it possible to deny users from doing docker pull/push from docker hub but they can use private registries ?

Thanks,
Charan

**Comment by tsandall** [comment]
Yes, you can do this. Assuming you have a well-known private registry URL you can do something like... `startswith(input.Body.Image, "myregistry.com/")`. You could make it more clever with a whitelist or you could try using regex.

**Comment by CharanBR** [comment]
thanks for the reply. I'm doing this as you suggested.

```
deny {
  private_registry
}
private_registry {
    input.Body.Image[_] = registry
    not startswith(registry, "mvnrepo")
}
```
So, this should work when I run commands like `docker run centos /bin/bash` but the commands executes runs successfully without interruption from opa. Also, how can to stop if I pull/push like
`docker pull centos`

**Comment by tsandall** [comment]
I think the `input.Body.Image[_] = registry` line should be changed to `input.Body.Image = registry`. The image field is just a string so `input.Body.Image[_]` is not going to do what you want it to (it'll be undefined because you can't iterate over strings in OPA.)

If you want to restrict pull/push you can, but I think you'll need to look at the input.Path (those commands generate simple HTTP POST calls where the image name is in the path.)

To figure out what the input looks like I'd recommend running the plugin and looking at the logs/stdout. The plugin prints the input that it provides to OPA. You can figure out what the input looks like and then write the policy accordingly.

Hope this helps.

**Comment by vijaychandranimmakuri** [comment]
Hi @cbrdy did you got any solution for that?
